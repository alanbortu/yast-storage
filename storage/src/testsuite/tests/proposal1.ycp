{
    include "testsuite.ycp";

    map READ = $[
	"probe" : $[
	    "architecture" : "i385",
	    "cdrom" : []
	    ],
	"proc" : $[
	    "swaps" : []
	],
        "sysconfig" : $[
	    "storage" : $[
		"DEFAULT_FS" : "ext4"
	    ],
	    "bootloader" : $[
		"LOADER_TYPE" : "grub"
	    ],
            "language" : $[
                "RC_LANG"               : "en_US.UTF-8",
                "RC_LC_MESSAGES"        : "",
            ],
        ],
        "target" : $[
            "size"              : 0,
            "bash_output"       : $[],
            "yast2"             : $[],
            "dir"               : [],
        ],
    ];

    TESTSUITE_INIT([READ, $[], READ], nil);


    import "Storage";
    import "StorageProposal";
    import "ProductFeatures";

    ProductFeatures::SetBooleanFeature("partitioning", "try_separate_home", true);
    ProductFeatures::SetBooleanFeature("partitioning", "proposal_lvm", false);
    ProductFeatures::SetStringFeature("partitioning", "root_base_size", "25 GB");

    StorageProposal::GetControlCfg();

    map<string, map> target_map = Storage::GetTargetMap();
    map<string, any> prop = StorageProposal::get_inst_prop(target_map);

    if (prop["ok"]:false)
    {
	Storage::SetTargetMap(prop["target"]:$[]);

	list<map> infos = Storage::GetCommitInfos();

	DUMP("Proposal:");
	foreach(map info, infos, {
	    string text = info[`text]:"";
	    if (info[`destructive]:false)
		text = text + " [destructive]";
	    DUMP(text);
	});
    }
    else
    {
	DUMP("No proposal.");
    }
}
