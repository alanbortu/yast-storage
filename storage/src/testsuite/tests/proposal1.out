Dump	Proposal:
Dump	Create extended partition /dev/sda3 (147.09 GB)
Dump	Create swap partition /dev/sda5 (2.01 GB)
Dump	Create root partition /dev/sda6 (20.00 GB) with ext4
Dump	Create partition /dev/sda7 (125.08 GB) for /home with ext4
